data/membership-schema/src/main/resources/net/corda/schema/membership/member/dynamic/registration/1.0/corda.member.dynamic.registration.json

```diff
@@ -36,6 +36,11 @@
     "^corda.ledger.keys.[0-9]+.signature.spec$": {
       "description": "Optional. The ledger public key signature spec. Can be multiple specified by index.",
       "type": "string"
+    },
+    "(?=^(?!corda\\.).+)(?=^(?=.{1,128}$)[a-zA-Z0-9]*[^$%^&*;:,<>?()\"']*$)": {
```

Comment:
This regex is a bit scary.. I am a bit worried it might be a maintenance pain in the future. It made me wonder; should we reconsider and instead limit custom attributes under a specific prefix, say `ext.`?

Reply:
that's true. I see we have the checks on the verified performed on the member side as well, so we could consider removing the validation from the schema if it makes things a bit easier to understand and maintain. 

Having a specific namespace for custom properties sounds attractive anyway though, since we'll be able to reason better about where they are. For example, if we want to evolve it further in the future (e.g. add extra category of fields), we won't have to adjust the logic that filters them out. It feels a bit more robust. I think we've discussed that in the past, but I can't recall if there was an important reason we didn't go for it.

Reply:
Not sure there was an important reason. I was thinking previously that using the `corda` namespace as our own protected namespace, and allowing the cordapp developers to define their own namespace separate to that made sense rather than having the cordapp developer namespace as a subset of the corda namespace. So they could have keys like `company.myKey` (where `company` could be the name of the company developing the cordapp) instead of `corda.ext.company.myKey`. I'm not strongly opposed to using `corda.ext` though.

Reply:
One thing to note is this schema is only checked on the member side and we can have infinite many `corda.` fields. Maybe we should restrict the total number of fields or the total size in the member context.

Reply:
I added https://r3-cev.atlassian.net/browse/CORE-11768